packages/js/product-editor/src/components/header/hooks/use-preview/use-preview.tsx

```diff
@@ -14,29 +14,22 @@ import { MouseEvent } from 'react';
  */
 import { useValidations } from '../../../../contexts/validation-context';
 import { WPError } from '../../../../utils/get-product-error-message';
+import { PreviewButtonProps } from '../../preview-button';
 
 export function usePreview( {
+	productId,
+	productStatus,
 	disabled,
 	onClick,
 	onSaveSuccess,
 	onSaveError,
 	...props
-}: Omit< Button.AnchorProps, 'aria-disabled' | 'variant' | 'href' > & {
+}: PreviewButtonProps & {
 	onSaveSuccess?( product: Product ): void;
 	onSaveError?( error: WPError ): void;
 } ): Button.AnchorProps {
 	const anchorRef = useRef< HTMLAnchorElement >();
 
-	const [ productId ] = useEntityProp< number >(
```

Comment:
Same thing, I was trying to avoid side effects also considering performance. I choose to get the id once and share it across the 3 inner components instead of having a side effect in each component to get the same value. You will say that the answer is being cached, but the fastest request is the one you don't have to do.

Reply:
>You will say that the answer is being cached, but the fastest request is the one you don't have to do.

You know me well 😄 

I don't think we'll get much measurable performance out of this and believe we may be prematurely optimizing this.

I can understand if we're removing the hooks to make this component more presentational and easier to test, but we're still using `useEntityProp` in this component below.  I'd prefer to either have all those hooks removed or keep the logic in here instead of some of it passed via props and some via hooks.